symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -218,6 +248,56 @@ public V4Message send(@Nonnull String streamId, @Nonnull String message) {
     ));
   }
 
+  /**
+   * Sends a message to the stream ID passed in parameter.
+   *
+   * @param streamId    the ID of the stream to send the message to
+   * @param message     the message to send to the stream
+   * @return a {@link V4Message} object containing the details of the sent message
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-message-v4">Create Message v4</a>
+   */
+  public V4Message send(@Nonnull String streamId, @Nonnull Message message) {
+    File tempFile;
+    Attachment attachment = message.getAttachment();
+    if (attachment != null) {
+      try {
+        String[] fileNameArrays = attachment.fileName().split("\\.");
+        if (fileNameArrays.length < 2) {
+          throw new MessageCreationException("Invalid attachment's file name.");
+        }
+        tempFile = File.createTempFile(fileNameArrays[0], "." + fileNameArrays[fileNameArrays.length - 1]);
```

Comment:
I would put `fileNameArrays[1]` since you validated the length above. In my opinion putting index 1 is clearer.

Reply:
I am not sure if the user passes a name with multiple `.` like `file.name.ext`